clickhouse/client.cpp

```diff
@@ -296,73 +320,100 @@ void Client::Impl::Ping() {
 }
 
 void Client::Impl::ResetConnection() {
+    connected_host_port_.reset();
+    for (int i = -1; i < int(options_.hosts_ports.size()); ++i) {
+        const ClientOptions::HostPort& host_port = i == -1 ? ClientOptions::HostPort(options_.host) : options_.hosts_ports[i];
```

Comment:
Please update the list of endpoints before copying it to a `options_` so you don't have to do this. (see above)